### ISSUE TITLE
destroy method on menu (dropdown and accordion) doesn't work

### ISSUE THREAD

**Issue by geosenna** [zurb/foundation-sites#11456]
The method to destroy a menu (dropdown or accordion) is not working in my project.

```
var nav = $('.js-nav'),
    menuDropdown,
    menuAccordion;

if (Foundation.MediaQuery.atLeast('large')) {
    //initialize the dropdown menu if desktop
    menuDropdown = new Foundation.DropdownMenu(nav, navDropdownOptions);
} else {
    //initialize the accordion menu if tablet or mobile
    menuAccordion = new Foundation.AccordionMenu(nav, navAccordionOptions);
}

//event when media query change
$(window).on('changed.zf.mediaquery', function(event, newSize, oldSize) {
    if (newSize == 'small' || newSize == 'medium') {
        if (menuDropdown) menuDropdown.Foundation('_destroy');
        menuAccordion = new Foundation.AccordionMenu(nav, navAccordionOptions);
    } else {
        if (menuAccordion)  menuAccordion.Foundation('_destroy');
        menuDropdown = new Foundation.DropdownMenu(nav, navDropdownOptions);
    }
});
```

Insetad I get the folowwing issue:
`app.js:49 Uncaught TypeError: f.Foundation is not a function`

I try several syntax such as:
```
 menuAccordion.Foundation('_destroy');
 menuAccordion.foundation('_destroy');
 nav.Foundation('_destroy');
 nav.foundation('_destroy');
```

the one below doesn't bring any error, but doesn't destroy the menu
`menuAccordion.destroy();`

Am i missing something here?

**Comment by ncoden** [comment]
@geosenna Take a look at [Programmatic Use](https://foundation.zurb.com/sites/docs/javascript.html#programmatic-use) in the Foundation documentation.

> Most plugins have a public API that allows you to manipulate it through JavaScript. Refer to a plugin's documentation to learn what functions are available. Invoking methods is easy as pie:
>
> ```js
> $('#reveal').foundation('open'); //will open a Reveal modal with id `reveal`.
> ```
>
> ```js
> $('[data-tabs]').eq(0).foundation('selectTab', $('#example')); //will change the > first Tabs on the page to whatever panel you choose.
> ```
> 
> ```js
> $('.tooltip').foundation('_destroy'); //will destroy all Tooltips on the page.
>

When creating a component dynamically with `new Foundation.DropdownMenu(...)`, it returns you get a plugin instance that does not have anything like a `Foundation` property. You can destroy it with the `.destroy()` method.

When using jQuery to initialize components with `$(document).foundation();`, you initialized all component in `document` and do not have access to their instances. To control (and destroy) your plugins, you have to continue using jQuery with `$('#yourDropdownMenu').foundation('_destroy');`.

There no such method like `mycomponent.Foundation`. There is only:
- the global `Foundation` object you can use to dynamically create plugins.
- the jQuery `$.foundation()` interface you can use to initialize all plugins in a particular DOM node..

I just tested `mycomponent.destroy()` and `$('#mycomponent').foundation('_destroy')`, and it works fine.

When you "destroy" a component, it will only remove all the Foundation datas and events associated to it. The HTML is still there as this is _your_ HTML, and you can remove it manually if you want to.

If this is not your issue, please provide a CodePen so we can better understand your issue, and most of all reproducing it.

**Comment by ncoden** [comment]
Closing as this seems to come from a wrong usage of the framework.

**Comment by DanielRuf** [comment]
> It appears that when I destroy a dropdown menu and replace it by a accordion menu, the class .is-dropdown-submenu-item is still here.
> 
> So my custom style applied on this class is in conflict when the accordion is set.
> 
> I created a CodePen: codepen.io/anon/pen/MqYqYR?editors=1111
> 
> The class .is-dropdown-submenu-item is still has a red color applied. Even when I destroy the dropdown and add a accordion instead, the class (and the red color) are still here.

Can you create a new issue for this?

Also I think we solved this in 6.5.0 (RC 1 or 2).

**Comment by DanielRuf** [comment]
> It appears that when I destroy a dropdown menu and replace it by a accordion menu, the class .is-dropdown-submenu-item is still here.
> 
> So my custom style applied on this class is in conflict when the accordion is set.
> 
> I created a CodePen: codepen.io/anon/pen/MqYqYR?editors=1111
> 
> The class .is-dropdown-submenu-item is still has a red color applied. Even when I destroy the dropdown and add a accordion instead, the class (and the red color) are still here.

Can you create a new issue for this?
